Return case insensitive check

DmitriiBezborodnikov opened the pull request:
Return insensitive check after #8453 

Problem: `->andWhere("u.name = ?1 or u.username = ?1");` did not wrap part in parenthesis when `or` or `and` was written in lowercase anymore. It still worked for uppercase `OR` and `AND`.

Fixes #8595

DmitriiBezborodnikov committed:
Return case insensitive check

greg0ire commented:
@Warxcell please review

Warxcell commented:
Yeah, seems we missed the case with lower case keywords. And there were no tests that could catch it. :(

greg0ire commented:
I tried restoring the source code and running the test suite, and the test suite did not fail, because the test class, is not suffixed with `Test` :see_no_evil:

Warxcell commented:
> I tried restoring the source code and running the test suite, and the test suite did not fail, because the test class, is not suffixed with `Test`

./vendor/bin/phpunit tests/Doctrine/Tests/ORM/Functional/QueryBuilderParenthesis.php 
PHPUnit 9.5.2 by Sebastian Bergmann and contributors.

..                                                                  2 / 2 (100%)

Time: 00:00.063, Memory: 12.00 MB

OK (2 tests, 4 assertions)


So phpunit have different behaviour depending on how you run it? That's bad :( It should tell me that this is not valid test IMO.

greg0ire commented:
Yup… kinda weird that my last commit does not result in an increase in coverage.

I have to go for now but I've started working on a `find` expression to find more test cases like this: `find tests -type f -name '*.php' -not \( -name '*Test.php' -o -wholename 'tests/Doctrine/Tests/Models/*' -o -wholename 'tests/Doctrine/Tests/Proxies/__CG__*' -o -wholename 'tests/Doctrine/Tests/ORM/Tools/Export/*' \)`

greg0ire committed:
Drop create table at shutdown

It makes tests more isolated from each other: another test relying on
some tables including some of the ones created here may fail creating
the tables it needs because they already exist.

greg0ire committed:
Make sure tests are suffixed with Test

They will not be taken into account when running vendor/bin/phpunit
otherwise.

greg0ire commented:
Let's treat that one separately, I'll open another PR for it.

greg0ire commented:
Thanks @DmitriiBezborodnikov !
